src/middlewared/middlewared/plugins/afp.py

```diff
@@ -39,3 +42,214 @@ class Config:
             await self._update_service(old, new)
 
         return new
+
+
+class SharingAFPService(CRUDService):
+    class Config:
+        namespace = 'sharing.afp'
+        datastore = 'sharing.afp_share'
+        datastore_prefix = 'afp_'
+
+    @accepts(Dict(
+        'sharingafp_create',
+        Str('path'),
+        Bool('home'),
+        Str('name'),
+        Str('comment'),
+        Str('allow'),
+        Str('deny'),
+        Str('ro'),
+        Str('rw'),
```

Comment:
Seems like these 4 above should also be a List to make API more friendly and ease validation.